### ISSUE TITLE
Add a submit button to the search bar

### ISSUE THREAD

**Issue by erquhart** [netlify/netlify-cms#302]
Because it's the right thing to do.

**Requirements**
* Add a "Search" button to submit the search form in the app header
* Use the standard button from React Toolbox
* We should probably update the color of the button to the primary brand color, as used in the header text
* We may as well update all primary buttons in the CMS to use that color, but this can be split off if it's out of the way to do so

**Comment by tech4him1** [comment]
@erquhart Can you explain why this was closed? I still think it is unclear that you have to push enter to search.

**Comment by erquhart** [comment]
Busted. Maybe we just shouldn't have an enter button at all, just a 1 second debounce (in case an integration is in use).

**Comment by tech4him1** [comment]
I would be cool with an instant search because of the small usable space we have, although I still think that is less clear.

**Comment by erquhart** [comment]
Ideally I think it would look like this for all backends:

* A limited number of results appear in a dropdown directly beneath the search input while typing
* Pressing enter or clicking submit takes you to a full results view

**Comment by mittalyashu** [comment]
Why we need a "Search" button?

Even the GitHub doesn't have a search button.

![2018-09-04_23-39-22](https://user-images.githubusercontent.com/29014463/45049422-2d86dd00-b09c-11e8-81ab-b026b71b6f55.gif)

**Comment by erquhart** [comment]
Many interfaces don't, it's a fair point.

**Comment by tech4him1** [comment]
I think we should have one of two things here:
 - Instant search
 - A search button

Right now, if I go to type in the search box and don't think to press "Enter", I get nothing: no feedback at all on what do do.

**Comment by mittalyashu** [comment]
I agree with @tech4him1, we should go for `instant search` instead of a search button. As every user want the information ASAP, without doing much work and that's why instant search will be more useful.

As we are discussing search, how about we add Algolia, where users can add API keys and search bar will be replaced with Algolia search bar. Where users can not only search the title, but also the content inside each post.

**Comment by erquhart** [comment]
Instant search is a more intuitive approach. Submission is implicit, which is why I created this issue in the first place, but it's a common enough pattern that it shouldn't be a problem.

There's a custom Algolia integration already in the CMS, it just isn't a public feature yet. That's pending backend/integration API redesign work that's currently underway.
